File: eTouristAgencyAPI/eTouristAgencyAPI.Services/Database/Models/OfferDiscount.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace eTouristAgencyAPI.Services.Database.Models;

[Table("OfferDiscount")]
public partial class OfferDiscount
{
    [Key]
    public Guid Id { get; set; }

    public Guid DiscountTypeId { get; set; }

    [Column(TypeName = "decimal(15, 2)")]
    public decimal Discount { get; set; }

    public Guid OfferId { get; set; }

    public DateTime ValidFrom { get; set; }

    public DateTime ValidTo { get; set; }

    public DateTime CreatedOn { get; set; }

    public DateTime ModifiedOn { get; set; }

    public Guid CreatedBy { get; set; }

    public Guid ModifiedBy { get; set; }

    [ForeignKey("CreatedBy")]
    [InverseProperty("OfferDiscountCreatedByNavigations")]
    public virtual User CreatedByNavigation { get; set; } = null!;

    [ForeignKey("DiscountTypeId")]
    [InverseProperty("OfferDiscounts")]
    public virtual EntityCodeValue DiscountType { get; set; } = null!;

    [ForeignKey("ModifiedBy")]
    [InverseProperty("OfferDiscountModifiedByNavigations")]
    public virtual User ModifiedByNavigation { get; set; } = null!;

    [ForeignKey("OfferId")]
    [InverseProperty("OfferDiscounts")]
    public virtual Offer Offer { get; set; } = null!;

    [InverseProperty("OfferDiscount")]
    public virtual ICollection<Reservation> Reservations { get; set; } = new List<Reservation>();
}

File: Frontend/etouristagency_desktop/lib/models/offer_discount/offer_discount.dart
import 'package:etouristagency_desktop/models/entity_code_value/entity_code_value.dart';

class OfferDiscount {
  String? id;
  String? discountTypeId;
  double? discount;
  DateTime? validFrom;
  DateTime? validTo;
  EntityCodeValue? discountType;
  bool? isEditable;
  bool? isRemovable;

  OfferDiscount({
    this.id,
    this.discountTypeId,
    this.discount,
    this.validFrom,
    this.validTo,
    this.discountType,
    this.isEditable,
    this.isRemovable,
  });

  factory OfferDiscount.fromJson(Map<String, dynamic> json) {
    return OfferDiscount(
      id: json["id"],
      discountTypeId: json["discountTypeId"],
      discount: json["discount"],
      validFrom: json["validFrom"] != null
          ? DateTime.parse(json["validFrom"])
          : null,
      validTo: json["validTo"] != null ? DateTime.parse(json["validTo"]) : null,
      discountType: json["discountType"] != null
          ? EntityCodeValue.fromJson(json["discountType"])
          : null,
      isEditable: json["isEditable"],
      isRemovable: json["isRemovable"],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "discountTypeId": discountTypeId,
      "discount": discount?.toString(),
      "validFrom": validFrom,
      "validTo": validTo,
    };
  }
}
